lib/openstax/exchange/fake_client/fake_client.rb

```diff
@@ -43,20 +43,18 @@ def token
       end
 
       def create_identifier
-        SecureRandom.hex(64)
+        {
+          'read' => SecureRandom.hex(64),
+          'write' => SecureRandom.hex(64)
+        }
       end
 
       def record_multiple_choice_answer(identifier, resource, trial, answer)
-        @multiple_choice_responses[identifier] ||= {}
-        @multiple_choice_responses[identifier][resource] ||= {}
-
-        raise "invalid resource" \
-          unless URI(resource).host == "exercises.openstax.org"
-
-        raise "duplicate response for (identifier,resource,trial) triplet" \
-          if @multiple_choice_responses[identifier][resource][trial]
+        raise "invalid resource" unless host.ends_with?("openstax.org") || \
+                                        host.ends_with?("localhost")
 
-        @multiple_choice_responses[identifier][resource][trial] = answer
+        @multiple_choice_responses[identifier][resource][trial] ||= []
+        @multiple_choice_responses[identifier][resource][trial] << answer
```

Comment:
What ensures that `@multiple_choice_responses[identifier]` and `@multiple_choice_responses[identifier][resource]` are initialized?